### DISCUSSION TITLE
ImageRepository connectivity DigitalOcean container registry

### DISCUSSION THREAD

**Discussion by ikavvouras** [fluxcd/flux2#4105] [Q&A] [answered]
Hello, I've tried to configure ImageRepository for a container that resides on DigitalOcean container registry with several ways but it constantly fails with: 
```
flux create image repository app1-repo \
  --secret-ref flux-registry \
  --image registry.digitalocean.com/__organiztion-name__/app1 --interval 5m
✚ generating ImageRepository
► applying ImageRepository
✗ failed to get API group resources: unable to retrieve the complete list of server APIs: image.toolkit.fluxcd.io/v1beta2: the server could not find the requested resource
```

I've tried to declare secret-ref with either with username/password or docker
```
kubectl create secret --namespace=flux-system generic flux-registry \                         
  --from-file=.dockerconfigjson=docker-config.json \
  --type=kubernetes.io/dockerconfigjson
```
or with the default secret that k8s uses to pull images, but it still fails with the aforementioned error.

**Comment by makkes** [comment]
Does the ImageRepository CRD even exist on your cluster? Check with

```
k get crd imagerepositories.image.toolkit.fluxcd.io
```

The image automation CRDs (including ImageRepository) aren't installed by default. Look at the `--components-extra` flag in the [the bootstrap options](https://fluxcd.io/flux/cmd/flux_bootstrap/#options).

**Reply by ikavvouras** [reply, marked as answer]
That was it, I installed image reflector and automation controller. Thank you!

```
flux install --components-extra="image-reflector-controller,image-automation-controller"
```
